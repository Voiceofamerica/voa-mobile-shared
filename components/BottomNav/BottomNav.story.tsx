
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import ArticleBlurb from '../../types/ArticleBlurb'

import BottomNav, { IconItem, RoundItem } from './'

const clickAction = action('IconItem.onClick()')
const roundClickAction = action('RoundItem.onClick()')

storiesOf('BottomNav', module)
  .add('with round item', () => (
    <BottomNav>
      <IconItem onClick={clickAction}>AB</IconItem>
      <IconItem onClick={clickAction}>CD</IconItem>
      <RoundItem onClick={roundClickAction}>EF</RoundItem>
      <IconItem onClick={clickAction}>GH</IconItem>
      <IconItem onClick={clickAction}>IJ</IconItem>
    </BottomNav>
  ))
  .add('without round item', () => (
    <BottomNav>
      <IconItem onClick={clickAction}>AB</IconItem>
      <IconItem onClick={clickAction}>CD</IconItem>
      <IconItem onClick={clickAction}>EF</IconItem>
      <IconItem onClick={clickAction}>GH</IconItem>
      <IconItem onClick={clickAction}>IJ</IconItem>
    </BottomNav>
  ))
  .add('with active item', () => (
    <BottomNav>
      <IconItem onClick={clickAction}>AB</IconItem>
      <IconItem onClick={clickAction} active>CD</IconItem>
      <RoundItem onClick={roundClickAction}>EF</RoundItem>
      <IconItem onClick={clickAction}>GH</IconItem>
      <IconItem onClick={clickAction}>IJ</IconItem>
    </BottomNav>
  ))
  .add('with active round item', () => (
    <BottomNav>
      <IconItem onClick={clickAction}>AB</IconItem>
      <IconItem onClick={clickAction}>CD</IconItem>
      <RoundItem onClick={roundClickAction} active>EF</RoundItem>
      <IconItem onClick={clickAction}>GH</IconItem>
      <IconItem onClick={clickAction}>IJ</IconItem>
    </BottomNav>
  ))
