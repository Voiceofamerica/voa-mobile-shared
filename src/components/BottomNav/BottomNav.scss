
@import "../../commonStyles/index.scss";

$round-item-offset: calc(-0.7vh + -10px);
$round-item-size: calc(2.5vh + 50px);

.bottomNav {
  height: $menu-height;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  background: $light-grey;
  border-top: 1px solid $border-grey;
  display: flex;
  flex-direction: row;
  font-size: $menu-font-size;
  overflow-y: show;

  .iconItem {
    flex-grow: 1;
    background: none;
    font-weight: bold;
    border: none;
    border-radius: 0;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-content: center;
    align-items: center;
    justify-content: center;
    outline: none;
    font-size: 1em;

    &.activeItem {
      color: rgba(0, 0, 0, 0.3);
    }

    &:active {
      background: white;
    }
  }

  .roundItem {
    background: none;
    font-weight: bold;
    border: none;
    border-radius: 0;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-content: center;
    align-items: center;
    justify-content: center;
    outline: none;
    margin-top: $round-item-offset;
    width: $round-item-size;
    height: $round-item-size;
    border-radius: $round-item-size;
    box-shadow: 0 -3px 2px rgba(0, 0, 0, 0.2);
    background: $light-grey;
    font-size: 1em;
    
    &.activeItem {
      color: rgba(0, 0, 0, 0.3);
    }

    &:active {
      background: white;
    }
  }
}
