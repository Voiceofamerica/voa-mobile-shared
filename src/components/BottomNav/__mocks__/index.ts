
import * as React from 'react'

export default (props) => (
  React.createElement('bottom-nav', { props }, props.children)
)

export const IconItem = (props) => (
  React.createElement('icon-item', { props }, props.children)
)

export const RoundItem = (props) => (
  React.createElement('round-item', { props }, props.children)
)
