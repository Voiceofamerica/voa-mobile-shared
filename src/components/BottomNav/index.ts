
export { default } from './BottomNav'
export { default as IconItem } from './IconItem'
export { default as RoundItem } from './RoundItem'
