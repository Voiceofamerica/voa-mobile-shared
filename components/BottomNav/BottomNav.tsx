
import * as React from 'react'
import { Subscription } from 'rxjs/Subscription'

import { bottomNav } from './BottomNav.scss'

export type Props = React.Props<any> & {}

function BottomNav ({ children }: Props) {
  return (
    <div className={bottomNav}>
      {
        children
      }
    </div>
  )
}

export default BottomNav
