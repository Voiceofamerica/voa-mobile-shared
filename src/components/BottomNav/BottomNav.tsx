
import * as React from 'react'

import { bottomNav, flexFlow } from './BottomNav.scss'

export type Props = React.Props<any> & {
  flex?: boolean
  style?: React.CSSProperties,
  className?: string,
}

function BottomNav ({ children, flex, className = '', style }: Props) {
  const fullClassName = flex
                      ? `${bottomNav} ${flexFlow} ${className}`
                      : `${bottomNav} ${className}`

  return (
    <div className={fullClassName} style={style}>
      {
        children
      }
    </div>
  )
}

export default BottomNav
