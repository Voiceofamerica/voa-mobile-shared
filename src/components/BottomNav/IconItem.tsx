
import * as React from 'react'

import { iconItem, activeItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
  className?: string,
  active?: boolean,
}

function IconItem ({ children, onClick = () => null, active = false, className = '' }: Props) {
  const iconClass = active
    ? `${iconItem} ${className} ${activeItem}`
    : `${iconItem} ${className}`

  return (
    <div className={iconClass} onClick={() => onClick()}>
      {
        children
      }
    </div>
  )
}

export default IconItem
