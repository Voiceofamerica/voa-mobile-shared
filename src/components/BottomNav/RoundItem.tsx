
import * as React from 'react'

import { roundItem, activeItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
  active?: boolean,
  style?: React.CSSProperties,
  className?: string,
}

function RoundItem ({ className = '', style, children, onClick = () => null, active = false }: Props) {
  const fullClassName = active
    ? `${roundItem} ${activeItem} ${className}`
    : `${roundItem} ${className}`
  return (
    <div className={fullClassName} style={style} onClick={() => onClick()}>
      {
        children
      }
    </div>
  )
}

export default RoundItem
