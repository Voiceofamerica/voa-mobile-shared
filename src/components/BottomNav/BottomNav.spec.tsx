
import * as React from 'react'
import { create } from 'react-test-renderer'

import BottomNav from './BottomNav'

describe('<BottomNav />', () => {
  describe('shapshots', () => {
    it('should render empty', () => {
      const element = create((
        <BottomNav />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given children', () => {
      const element = create((
        <BottomNav>
          <div>child1</div>
          <div>child2</div>
        </BottomNav>
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given style', () => {
      const element = create((
        <BottomNav style={{ top: 0, left: 0, right: 0, bottom: 0 }} />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given className', () => {
      const element = create((
        <BottomNav className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })
  })
})
