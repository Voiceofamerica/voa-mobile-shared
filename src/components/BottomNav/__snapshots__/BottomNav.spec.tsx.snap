// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomNav /> shapshots should render empty 1`] = `
<div
  className="bottomNav "
  style={undefined}
/>
`;

exports[`<BottomNav /> shapshots should render with the given children 1`] = `
<div
  className="bottomNav "
  style={undefined}
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</div>
`;

exports[`<BottomNav /> shapshots should render with the given className 1`] = `
<div
  className="bottomNav myClass"
  style={undefined}
/>
`;

exports[`<BottomNav /> shapshots should render with the given style 1`] = `
<div
  className="bottomNav "
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
/>
`;
